homeassistant/components/pilight.py

```diff
@@ -0,0 +1,116 @@
+"""
+Component to create an interface to a Pilight daemon (https://pilight.org/).
+
+Pilight can be used to send and receive signals from a radio frequency
+module (RF receiver).
+
+RF commands received by the daemon are put on the HA event bus.
+RF commands can also be send with a pilight.send service call.
+"""
+# pylint: disable=import-error
+import logging
+import socket
+
+import voluptuous as vol
+
+from homeassistant.helpers import validate_config
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.config_validation import ensure_list
+from homeassistant.const import EVENT_HOMEASSISTANT_START
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+REQUIREMENTS = ['pilight==0.0.2']
+
+DOMAIN = "pilight"
+EVENT = 'pilight_received'
+SERVICE_NAME = 'send'
+
+ATTR_PROTOCOL = 'protocol'
+
+# The pilight code schema depends on the protocol
+# Thus only require to have the protocol information
+RF_CODE_SCHEMA = vol.Schema({vol.Required(ATTR_PROTOCOL): cv.string},
```

Comment:
Yes. Pilight has plenty of [protocols](https://wiki.pilight.org/doku.php/protocols). You can also define your own. That means somebody has to frequently update this to all possible protocols. A [weather station](https://wiki.pilight.org/doku.php/alecto_wx500_v7_0) gives you for example a battery entry and a wind direction entry. A [dimmer gives](https://wiki.pilight.org/doku.php/arctech_dimmer_v7_0) you a dimmer level etc.